build/ci.go

```diff
@@ -390,6 +459,62 @@ func downloadLinter(cachedir string) string {
 	return filepath.Join(cachedir, base, "golangci-lint")
 }
 
+var errNoReleaseAvailable = errors.New("no release available")
+
+// downloadProtoc downloads the prebuilt protoc binary used to lint generated
+// files as a CI step.  It returns the full path of the 'protoc' executable or
+// an error.
+func downloadProtoc(cachedir string) (string, error) {
+	version := "27.1"
+	arch := runtime.GOARCH
+	operatingSystem := runtime.GOOS
+	var base string
+
+	csdb := build.MustLoadChecksums("build/checksums.txt")
+	version, err := build.Version(csdb, "protoc")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if runtime.GOOS == "windows" {
```

Comment:
I think this whole section would read easier if you used `switch` on GOOS/GOARCH.

Reply:
Hmmm, I will try to think about how to simplify this.  The difference in protoc archive naming scheme between windows releases and other architectures makes the logic here a bit convoluted.

Reply:
So yeah, we can of course use switch statements here.  But it won't take away from the weirdness of this part.